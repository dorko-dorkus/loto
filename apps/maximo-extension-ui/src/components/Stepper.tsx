import React from 'react';
import clsx from 'clsx';

export interface StepperProps {
  steps: string[];
  active: number;
  className?: string;
}

export default function Stepper({ steps, active, className }: StepperProps) {
  return (
    <ol className={clsx('flex gap-2', className)}>
      {steps.map((step, index) => (
        <li
          key={step}
          className={clsx(
            'rounded px-3 py-1',
            index === active
              ? 'bg-[var(--mxc-topbar-bg)] text-[var(--mxc-topbar-fg)]'
              : 'bg-[var(--mxc-sheet)] text-[var(--mxc-fg-subtle)]'
          )}
        >
          {step}
        </li>
      ))}
    </ol>
  );
}
