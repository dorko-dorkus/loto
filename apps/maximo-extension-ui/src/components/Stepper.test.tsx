import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import Stepper from './Stepper';

test('matches snapshot', () => {
  const { container } = render(
    <Stepper steps={['Step 1', 'Step 2', 'Step 3']} active={1} />
  );
  expect(container.firstChild).toMatchSnapshot();
});
