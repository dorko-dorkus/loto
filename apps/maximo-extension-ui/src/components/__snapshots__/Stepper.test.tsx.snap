// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<ol
  class="flex gap-2"
>
  <li
    class="rounded px-3 py-1 bg-[var(--mxc-sheet)] text-[var(--mxc-fg-subtle)]"
  >
    Step 1
  </li>
  <li
    class="rounded px-3 py-1 bg-[var(--mxc-topbar-bg)] text-[var(--mxc-topbar-fg)]"
  >
    Step 2
  </li>
  <li
    class="rounded px-3 py-1 bg-[var(--mxc-sheet)] text-[var(--mxc-fg-subtle)]"
  >
    Step 3
  </li>
</ol>
`;
